File: SisVendedoras/scripts/app/marca/MarcaController.js
app.controller("MarcaController", function ($scope, $window, MarcaServico, toastr) {
    $scope.gridOptions = {
        data: [],
        urlSync: true,
        pagination: {
            itemsPerPage: '10'
        }
    };

    $scope.carregarMarca = function () {
        var idMarca = $window.location.pathname.split('/')[3];

        MarcaServico.carregarMarca(idMarca).then(function (data) {
            if (data.marca !== undefined && data.erros === undefined) {
                $scope.marca = data.marca;
            }
            else {
                toastr.error(data.erros, 'Erro');
            }
        });
    };

    $scope.carregarMarcas = function () {
        MarcaServico.carregarMarcas().then(function (data) {
            if (data.lstMarcas !== undefined && data.erros === undefined) {
                $scope.lstMarcas = data.lstMarcas;
                $scope.gridOptions.data = data.lstMarcas;
            }
            else {
                toastr.error(data.erros, 'Erro');
            }
        });
    };

    $scope.cadastrar = function () {
        $scope.subimeter = true;

        if ($scope.frmCadastrar.$valid) {
            MarcaServico.cadastrar($scope.Descricao).then(function (data) {
                if (data.erros === undefined) {
                    toastr.success(data.mensagem, 'Sucesso');

                    $scope.subimeter = false;
                }
                else {
                    toastr.error(data.erros, 'Erro');

                    $scope.subimeter = false;
                }

                if (data.urlRedirecionar !== undefined) {
                    $window.location.href = data.urlRedirecionar;
                }
            });
        }
    };

    $scope.editar = function () {
        $scope.subimeter = true;

        if ($scope.frmEditar.$valid) {
            MarcaServico.editar($scope.marca).then(function (data) {
                if (data.erros === undefined) {
                    toastr.success(data.mensagem, 'Sucesso');

                    $scope.subimeter = false;
                }
                else {
                    toastr.error(data.erros, 'Erro');

                    $scope.subimeter = false;
                }

                if (data.urlRedirecionar !== undefined) {
                    $window.location.href = data.urlRedirecionar;
                }
            });
        }
    };

    $scope.deletar = function (marca) {
        MarcaServico.deletar(marca).then(function (data) {
            if (data.erros === undefined) {
                toastr.success(data.mensagem, 'Sucesso');
            }
            else {
                toastr.error(data.erros, 'Erro');
            }

            if (data.urlRedirecionar !== undefined) {
                $window.location.href = data.urlRedirecionar;
            }
        });
    };
});

File: SisVendedoras/Controllers/MarcaController.cs
using System;
using System.Web.Mvc;
using SisVendedoras.Authentication.Authentication;
using SisVendedoras.Infra.Repositorios;
using SisVendedoras.Servicos.MarcaServico;
using SisVendedoras.Dominio.Entidades;
using System.Linq;

namespace SisVendedoras.Controllers
{
    public class MarcaController : Controller
    {
        // GET: Marca
        public ActionResult Index()
        {
            return View();
        }

        public ActionResult Cadastrar()
        {
            return View();
        }

        [Route("EditarRegistro/{idMarca}")]
        public ActionResult EditarRegistro(long? idMarca)
        {
            return View();
        }

        public JsonResult CarregarMarca(long idMarca)
        {
            try
            {
                if (AuthenticationUser.UserSession != null)
                {
                    return Json(new { marca = new MarcaServico().GetById(idMarca) }, JsonRequestBehavior.AllowGet);
                }
                else
                {
                    return Json(new { urlRedirecionar = "/Login/Index" }, JsonRequestBehavior.AllowGet);
                }
            }
            catch (Exception)
            {

                throw;
            }
        }

        public JsonResult CarregarMarcas()
        {
            try
            {
                if (AuthenticationUser.UserSession != null)
                {
                    return Json(new { lstMarcas = new MarcaServico().GetAll().Where(tmp => tmp.IdUsuario.Equals(AuthenticationUser.UserSession.IdUsuario)) }, JsonRequestBehavior.AllowGet);
                }
                else
                {
                    return Json(new { urlRedirecionar = "/Login/Index" }, JsonRequestBehavior.AllowGet);
                }
            }
            catch (Exception ex)
            {
                return Json( new { erros = ex.Message });
            }
        }

        public JsonResult Salvar(string descricao)
        {
            try
            {
                if (AuthenticationUser.UserSession != null)
                {
                    new MarcaServico().Salvar(new Marca { IdUsuario = AuthenticationUser.UserSession.IdUsuario, Descricao = descricao });

                    return Json(new { mensagem = "Cadastro realizado com sucesso", urlRedirecionar = "/Marca/Index" });
                }
                else
                {
                    return Json(new { urlRedirecionar = "/Login/Index" });
                }
            }
            catch (Exception ex)
            {
                return Json(new { erros = ex.Message });
            }
        }

        public JsonResult Editar(Marca objMarca)
        {
            try
            {
                if (AuthenticationUser.UserSession != null)
                {
                    new MarcaServico().Editar(objMarca);

                    return Json(new { mensagem = "Editado com sucesso", urlRedirecionar = "/Marca/Index" });
                }
                else
                {
                    return Json(new { urlRedirecionar = "/Login/Index" });
                }
            }
            catch (Exception ex)
            {
                return Json(new { erros = ex.Message });
            }
        }

        public JsonResult Deletar(Marca objMarca)
        {
            try
            {
                if (AuthenticationUser.UserSession != null)
                {
                    new MarcaServico().Deletar(objMarca);

                    return Json(new { mensagem = "Deletado com sucesso", urlRedirecionar = "/Marca/Index" });
                }
                else
                {
                    return Json(new { urlRedirecionar = "/Login/Index" });
                }
            }
            catch (Exception ex)
            {
                return Json(new { erros = ex.Message });
            }
        }
    }
}